### PR TITLE
tests and move 2024 cleanup

### PR DESCRIPTION
**Pull request by dariorussi**
## Description 

Mostly moved tests in their own files. Particularly tests that have to be expanded.
Move 2024 cleanup.
Formatting and convention alignment.

This is almost no change except for few visibility changes we will talk about

## Test plan 

Existing test, no feature added

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: crates/sui-framework/packages/bridge/sources/treasury.move [hunk 1]
```diff
@@ -69,6 +69,21 @@ module bridge::treasury {
...
+
+    public fun decimal_multiplier<T>(self: &BridgeTreasury): u64 {
+        let metadata = self.get_token_metadata<T>();
+        metadata.decimal_multiplier
+    }
+
+    public fun notional_value<T>(self: &BridgeTreasury): u64 {
+        let metadata = self.get_token_metadata<T>();
+        metadata.notional_value
+    }
```
Comment (longbowlu) [inline crates/sui-framework/packages/bridge/sources/treasury.move:85]: do these need to be `public`? It looks like people won't be able to get `&BridgeTreasury` anyway: https://github.com/MystenLabs/sui/blob/feature/native-bridge/crates/sui-framework/packages/bridge/sources/bridge.move#L674

**Review by longbowlu — COMMENTED** [review]
overall looks good with a few nits

File: crates/sui-framework/packages/bridge/sources/committee.move [hunk 2]
```diff
@@ -84,15 +73,6 @@ module bridge::committee {
...
 
-    public(package) fun create(ctx: &TxContext): BridgeCommittee {
-        assert!(tx_context::sender(ctx) == @0x0, ENotSystemAddress);
-        BridgeCommittee {
-            members: vec_map::empty(),
-            member_registrations: vec_map::empty(),
-            last_committee_update_epoch: 0,
-        }
-    }
-
```
[the remark this answers is not in the document]
Comment (dariorussi) [reply crates/sui-framework/packages/bridge/sources/committee.move:95]: I really do not like that but I could add a public function marker

File: crates/sui-framework/packages/bridge/sources/committee.move [hunk 3]
```diff
@@ -170,24 +158,6 @@ module bridge::committee {
         emit(registration)
     }
 
-    // Assert if `bridge_pubkey_bytes` is duplicated in `member_registrations`.
-    // Dupicate keys would cause `try_create_next_committee` to fail and,
-    // in consequence, an end of epoch transaction to fail (safe mode run).
-    // This check will ensure the creation of the committee is correct.
-    fun check_uniqueness_bridge_keys(self: &BridgeCommittee, bridge_pubkey_bytes: vector<u8>) {
```
[the remark this answers is not in the document]
Comment (dariorussi) [reply crates/sui-framework/packages/bridge/sources/committee.move:177]: I do not think so and we are not yet 100% test coverage. So soon...

File: crates/sui-framework/packages/bridge/sources/treasury.move (hunk 1 above)
Comment (dariorussi) [reply crates/sui-framework/packages/bridge/sources/treasury.move:85]: I have not looked if they are used or not as they seemed accessors that are pretty harmless.
I can change them if we want

File: crates/sui-framework/packages/bridge/sources/message.move [hunk 4]
```diff
@@ -561,423 +554,83 @@ module bridge::message {
...
-    }
-
-    #[test]
-    fun test_peel_u64_be() {
-        let input = hex::decode(b"0000000000003039");
-        let expected = 12345u64;
-        let mut bcs = bcs::new(input);
-        assert!(peel_u64_be(&mut bcs) == expected, 0)
+    #[test_only]
+    public(package) fun validator_eth_addresses(block_list: &Blocklist): &vector<vector<u8>> {
```
[the remark this answers is not in the document]
Comment (dariorussi) [reply crates/sui-framework/packages/bridge/sources/message.move:633]: good call, I do not see usage so I will remove and see, thanks

**Comment by dariorussi** [comment]
feedback

**Review by patrickkuo — APPROVED** [review]
Thanks for the cleanup, LGTM

**Pull request merged**